### PR TITLE
Fix `ParagraphStyle` equals for layout cache

### PR DESCRIPTION
**Pull request by MatkovIvan**
Relax style comparison for layout cache to fix finding valid cached values

Fixes https://youtrack.jetbrains.com/issue/CMP-6879

### CODE REVIEW

File: modules/skparagraph/src/ParagraphCache.cpp [hunk 1]
```diff
@@ -190,7 +190,7 @@ bool ParagraphCacheKey::operator==(const ParagraphCacheKey& other) const {
 
     // There is no need to compare default paragraph styles - they are included into fTextStyles
```
Comment (pjBooms) [inline modules/skparagraph/src/ParagraphCache.cpp]: This comment became obsolete, I would remove it.

File: modules/skparagraph/src/ParagraphStyle.cpp [hunk 2]
```diff
@@ -24,6 +24,41 @@ TextIndent::TextIndent() {
...
+           this->fTextDirection == rhs.fTextDirection &&
+           this->fLinesLimit == rhs.fLinesLimit &&
+           this->fEllipsisUtf16 == rhs.fEllipsisUtf16 &&
+           this->fEllipsis == rhs.fEllipsis &&
+           nearlyEqual(fHeight, rhs.fHeight) &&
+           this->fTextHeightBehavior == rhs.fTextHeightBehavior &&
+           this->fTextIndent == rhs.fTextIndent &&
+           this->fFontRastrSettings == rhs.fFontRastrSettings &&
+           this->fHintingIsOn == rhs.fHintingIsOn &&
+           this->fReplaceTabCharacters == rhs.fReplaceTabCharacters &&
```
Comment (SergeevPavel) [inline modules/skparagraph/src/ParagraphStyle.cpp]: Can we extract some common part there to highlight the difference?

**Review by SergeevPavel — COMMENTED** [review]
Some test in `SkParagraphTest.cpp` that shows the change in behaviour would be really valuable

File: modules/skparagraph/src/ParagraphStyle.cpp (hunk 2 above)
Comment (pjBooms) [reply modules/skparagraph/src/ParagraphStyle.cpp]: Done

**Review by pjBooms — APPROVED** [review]

**Review by SergeevPavel — APPROVED** [review]

**Review by eymar — APPROVED** [review]

**Pull request merged**